cmd/telemetry-agent/main.go

```diff
@@ -164,12 +164,34 @@ func processMetrics(ctx context.Context, c config.Config, platformClient *platfo
 
 		metricsLogger := l.With(zap.String("file", pillarM.Filename))
 		platformCtx := platformLogger.GetContextWithLogger(ctx, metricsLogger.Desugar())
-		// send request to Percona Platform
-		if err := platformClient.SendTelemetry(platformCtx, "", report); err != nil {
-			if !errors.Is(err, context.Canceled) {
-				metricsLogger.Errorw("error during sending telemetry, will try on next iteration", zap.Error(err))
+		err := func() error {
+			funcCtx, cancel := context.WithTimeout(platformCtx, 60*time.Second)
+			defer cancel()
+			// send request to Percona Platform
+			return platformClient.SendTelemetry(funcCtx, "", report)
+		}()
```

Comment:
What is the reason to wrap this code in function like this? For defer?

Reply:
yes